engine/engine.go

```diff
@@ -315,3 +316,34 @@ func (e *Engine) pingWorker() bool {
 		return false
 	}
 }
+
+// apiGood returns true if the api can be reached and reports sufficient fil/cap to process tasks.
+func (e *Engine) apiGood(ctx context.Context) bool {
+	localBalance, err := e.node.WalletBalance(ctx, e.nodeConfig.WalletAddress)
+	if err != nil {
+		log.Errorw("could not query api for wallet balance", "error", err)
+		return false
+	}
+	if localBalance.LessThan(e.nodeConfig.MinWalletBalance) {
+		return false
```

Comment:
Even though this is not an error, perhaps logging a warning would be useful to alert admins that they need to rotate/fund the wallet.